### PR TITLE
Add implementation for cognitoidp admin_set_user_mfa_preference

### PR DESCRIPTION
**Pull request by linuxlewis**
* Update set_user_mfa_preference to handle PreferredMFA option
* Add tests

### CODE REVIEW

File: moto/cognitoidp/models.py [hunk 1]
```diff
@@ -1643,13 +1645,39 @@ def set_user_mfa_preference(
...
                 return None
         else:
             raise NotAuthorizedError(access_token)
 
+    def admin_set_user_mfa_preference(
+        self, user_pool_id, username, software_token_mfa_settings, sms_mfa_settings
+    ):
+        user = self.admin_get_user(user_pool_id, username)
+
+        if software_token_mfa_settings["Enabled"]:
```
Comment (bblommers) [inline moto/cognitoidp/models.py:1664]: The docs do not say whether this is required, so it's probably safer to assume this field may not be set:
```
if software_token_mfa_settings.get("Enabled", False):
```

File: moto/cognitoidp/models.py [hunk 2]
```diff
@@ -1643,13 +1645,39 @@ def set_user_mfa_preference(
...
+            if user.token_verified:
+                user.software_token_mfa_enabled = True
+            else:
+                raise InvalidParameterException(
+                    "User has not verified software token mfa"
+                )
+
+            if software_token_mfa_settings["PreferredMfa"]:
```
Comment (bblommers) [inline moto/cognitoidp/models.py:1672]: Same here

File: moto/cognitoidp/models.py [hunk 3]
```diff
@@ -1643,13 +1645,39 @@ def set_user_mfa_preference(
...
+                user.preferred_mfa_setting = "SOFTWARE_TOKEN_MFA"
+        elif sms_mfa_settings["Enabled"]:
```
Comment (bblommers) [inline moto/cognitoidp/models.py:1674]: Same here

File: moto/cognitoidp/models.py [hunk 4]
```diff
@@ -1643,13 +1645,39 @@ def set_user_mfa_preference(
...
+            user.sms_mfa_enabled = True
+
+            if sms_mfa_settings["PreferredMfa"]:
```
Comment (bblommers) [inline moto/cognitoidp/models.py:1677]: Same here

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @linuxlewis, would you mind adding a few more tests?
Ideally there's  a unit test for every possible path, especially a test where `SoftwareTokenMfaSettings` is set and another where we verify that the error is raised correctly.

The docs have a small example on what a negative test should look like:
http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html

**Comment by linuxlewis** [comment]
@bblommers Thanks for the review. Made updates to access parameters safely and added a negative test case. Please let me know if there's anything else to add

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the PR @linuxlewis

**Pull request merged**